src/diffusers/models/attention.py

```diff
@@ -249,13 +249,15 @@ def reshape_batch_dim_to_heads(self, tensor):
         return tensor
 
     def forward(self, hidden_states, context=None, mask=None):
-        batch_size, sequence_length, dim = hidden_states.shape
+        batch_size, sequence_length, _ = hidden_states.shape
 
         query = self.to_q(hidden_states)
         context = context if context is not None else hidden_states
         key = self.to_k(context)
         value = self.to_v(context)
 
+        dim = query.shape[-1]
```

Comment:
The `dim` should be computed with the projected `q`, `k`, `v` (i.e. `inner_dim`), not with `hidden_states`.

Reply:
The current CI passes, but the test has `inner_dim == hidden_states.shape[-1]` (both `64`). When this is not the case, we get an error.